### PR TITLE
Enhance gardener user

### PR DESCRIPTION
**Pull request by KristianZH**
**What this PR does / why we need it**:

Moves the logic from this PR in `gardener-extensions`: https://github.com/gardener/gardener-extensions/pull/372 here.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
@rfranzke @vpnachev 

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: pkg/operation/botanist/operatingsystemconfig.go [hunk 1]
```diff
@@ -161,12 +161,15 @@ func (b *Botanist) deployOperatingSystemConfigsForWorker(machineTypes []gardenco
...
 
+	sshKey := b.Secrets[v1alpha1constants.SecretNameSSHKeyPair].Data[secrets.DataKeySSHAuthorizedKeys]
+
 	originalConfig["osc"] = map[string]interface{}{
 		"type":                 machineImage.Name,
 		"purpose":              extensionsv1alpha1.OperatingSystemConfigPurposeReconcile,
 		"reloadConfigFilePath": common.CloudConfigFilePath,
 		"secretName":           secretName,
 		"customization":        customization,
+		"sshKey": string(sshKey),
```
Comment (vpnachev) [inline pkg/operation/botanist/operatingsystemconfig.go]:
```suggestion
		"sshKey":                  string(sshKey),
```

**Review by vpnachev — CHANGES_REQUESTED** [review]
Thanks!
Just a small formatting issue must be fixed.

**Review by rfranzke — DISMISSED** [review]

**Comment by vpnachev** [comment]
You might consider to remove the constant `OperatingSystemConfigUnitNameGardenerUserService` because it seems it is not used anywhere:
https://github.com/gardener/gardener/blob/c6f1ec73e444853971940a694185b65e20069dd5/pkg/apis/core/v1alpha1/constants/types_constants.go#L235-L236

**Comment by ialidzhikov** [comment]
@KristianZH, could you please rebase to get the `.ci/check` fix by @vpnachev ?

**Review by vpnachev — APPROVED** [review]
thanks!
/approve

**Pull request merged**